### PR TITLE
Emit config before parsing pipeline

### PR DESCRIPTION
**Pull request by hatchan**
And on invalid yaml

### CODE REVIEW

**Comment by Tonnu** [comment]
lgtm :+1:

File: cmd/runner.go [hunk 1]
```diff
@@ -454,12 +454,23 @@ func (p *Runner) SetupEnvironment(runnerCtx context.Context) (*RunnerShared, err
 	// Grab our config
 	rawConfig, stringConfig, err := p.GetConfig()
 	if err != nil {
```
Comment (mwhooker) [inline cmd/runner.go]: can this be simplified to one call to `Emit` before this line?

Comment (hatchan) [inline cmd/runner.go]: Changed as suggested

**Comment by hatchan** [comment]
Changed according to feedback, and added a check to see if `stringConfig` is not empty.

**Pull request merged**